plugins/org.eclipse.fordiac.ide.typemanagement/src/org/eclipse/fordiac/ide/typemanagement/refactoring/RefactorElementPropertyTester.java

```diff
@@ -31,9 +35,25 @@ protected boolean isSupported(final Object element) {
 			return isSupported(editPart.getModel());
 		}
 
-		return ((element instanceof final IInterfaceElement ie && !(element instanceof ErrorMarkerInterface))
-				&& (ie.getFBNetworkElement() instanceof final FB fb
-						&& !fb.getType().getTypeEntry().getFile().isReadOnly()))
-				|| (element instanceof final FB fbb && fbb.eContainer() instanceof BaseFBType);
+		if (element instanceof final EObject eObject
+				&& EcoreUtil.getRootContainer(eObject) instanceof final LibraryElement libraryElement
+				&& libraryElement.getTypeEntry().getFile().isReadOnly()) {
+			return false;
+		}
+
+		if (element instanceof final IInterfaceElement ie && !(ie instanceof ErrorMarkerInterface)) {
+			if (ie.getFBNetworkElement() != null && ie.getFBNetworkElement().getType() != null
+					&& ie.getFBNetworkElement().getType().getTypeEntry() != null
+					&& ie.getFBNetworkElement().getType().getTypeEntry().getFile() != null) {
+				return !ie.getFBNetworkElement().getType().getTypeEntry().getFile().isReadOnly();
+			}
+			if (ie.eContainer() instanceof final InterfaceList il && il.eContainer() instanceof final BaseFBType fbType
+					&& fbType.getTypeEntry() != null && fbType.getTypeEntry().getFile() != null) {
+				return !fbType.getTypeEntry().getFile().isReadOnly();
+			}
+			return false;
```

Comment:
This still returns false for interface elements not in the interface list of a BFB. Now that you have the check in line 38, you should be able to remove this check here and just return true, as was the original behavior:
```suggestion
			return true;
```